client/my-sites/media/main.jsx

```diff
@@ -291,8 +292,9 @@ class Media extends Component {
 
 		const selected =
 			selectedItems && selectedItems.length ? selectedItems : this.props.selectedItems;
+		const selectedIds = selected.map( ( { ID } ) => ID );
```

Comment:
The original `deleteMedia` action also filtered out items that don't have a (numeric) `ID` and I'm wondering whether that can happen. Transient media items, ones that are being uploaded, are shown in the media list but don't have a server-assigned `ID` yet. Their ID is some `transient-XX`-like string. Is it possible in the UI to select them and delete them?

Reply:
I think we could add a filter to be safe but as far as my testing went it wasn't possible to delete items which aren't fully uploaded yet. I tested that in `/media` not in the modal though. Will have a thorough look and probably add a check.